ports/stm32f4/common-hal/busio/I2C.c

```diff
@@ -32,23 +32,62 @@
 
 #include "shared-bindings/microcontroller/__init__.h"
 #include "supervisor/shared/translate.h"
+#include "stm32f4/periph.h"
 
-I2C_HandleTypeDef hi2c2;
+I2C_HandleTypeDef hi2c;
 
 void common_hal_busio_i2c_construct(busio_i2c_obj_t *self,
         const mcu_pin_obj_t* scl, const mcu_pin_obj_t* sda, uint32_t frequency, uint32_t timeout) {
     //TODO: Rework this entire section to use LL so we can properly assign pins
-    //      This will also be bundled with MSP removal
-    hi2c2.Instance = I2C1;
-    hi2c2.Init.ClockSpeed = 100000;
-    hi2c2.Init.DutyCycle = I2C_DUTYCYCLE_2;
-    hi2c2.Init.OwnAddress1 = 0;
-    hi2c2.Init.AddressingMode = I2C_ADDRESSINGMODE_7BIT;
-    hi2c2.Init.DualAddressMode = I2C_DUALADDRESS_DISABLE;
-    hi2c2.Init.OwnAddress2 = 0;
-    hi2c2.Init.GeneralCallMode = I2C_GENERALCALL_DISABLE;
-    hi2c2.Init.NoStretchMode = I2C_NOSTRETCH_DISABLE;
-    if(HAL_I2C_Init(&hi2c2) != HAL_OK) {
+    //      This will also be bundled with MSP removal  
+    I2C_TypeDef * I2Cx = NULL;
+    I2C_TypeDef * I2Cx_check = NULL;
+
+    for(uint i=0; i<(sizeof(mcu_i2c_list)/sizeof(*mcu_i2c_list));i++) {
+        if (mcu_i2c_list[i]->sda_pin == sda) {
+            switch (mcu_i2c_list[i]->i2c_index) {
+                case 1: I2Cx = I2C1;
+                        break;
+                #ifdef I2C2
+                case 2: I2Cx = I2C2;
+                        break;
+                #endif
+                #ifdef I2C3
+                case 3: I2Cx = I2C3;
+                        break;
+                #endif
+            }
+        } else if (mcu_i2c_list[i]->scl_pin == scl) {
+            switch (mcu_i2c_list[i]->i2c_index) {
+                case 1: I2Cx_check = I2C1;
+                        break;
+                #ifdef I2C2
+                case 2: I2Cx_check = I2C2;
+                        break;
+                #endif
+                #ifdef I2C3
+                case 3: I2Cx_check = I2C3;
+                        break;
+                #endif
+            }
+        }
+    }
+
+    if (I2Cx!=I2Cx_check || I2Cx==NULL) {
+        mp_raise_RuntimeError(translate("Invalid I2C pin selection"));
+    }
+    
+
+    hi2c.Instance = I2Cx;
+    hi2c.Init.ClockSpeed = 100000;
+    hi2c.Init.DutyCycle = I2C_DUTYCYCLE_2;
+    hi2c.Init.OwnAddress1 = 0;
+    hi2c.Init.AddressingMode = I2C_ADDRESSINGMODE_7BIT;
+    hi2c.Init.DualAddressMode = I2C_DUALADDRESS_DISABLE;
+    hi2c.Init.OwnAddress2 = 0;
+    hi2c.Init.GeneralCallMode = I2C_GENERALCALL_DISABLE;
+    hi2c.Init.NoStretchMode = I2C_NOSTRETCH_DISABLE;
+    if(HAL_I2C_Init(&hi2c) != HAL_OK) {
```

Comment:
Do you have to connect the pins to the peripheral using an internal mux? You'll probably need to track which alternate function each pin is on.

Reply:
That's a good point. I'll add it.